kcidev/subcommands/results.py

```diff
@@ -44,6 +47,45 @@ def fetch_tree_fast(origin):
     return fetch_from_api("tree-fast", params)
 
 
+def get_folder_repository(git_folder):
+    kci_msg("git folder: " + str(git_folder))
+    if git_folder:
+        current_folder = git_folder
+    else:
+        current_folder = os.getcwd()
+
+    previous_folder = os.getcwd()
+    os.chdir(current_folder)
+    dot_git_folder = os.path.join(current_folder, ".git")
+
+    # Check if we are in a git repository
+    if os.path.exists(dot_git_folder):
+        # Get remote origin url
+        git_config_path = os.path.join(dot_git_folder, "config")
+        git_config = configparser.ConfigParser()
+        git_config.read(git_config_path)
+        git_url = git_config.get('remote "origin"', "url")
+        # Get current branch name
+        process = subprocess.Popen(
+            ["git", "branch", "--show-current"], stdout=subprocess.PIPE, text=True
+        )
+        branch_name, branch_error = process.communicate()
+        branch_name = branch_name.strip()
+
+        # Get last commit hash
+        last_commit_hash_path = os.path.join(
+            dot_git_folder, "refs", "heads", branch_name
+        )
+        last_commit_hash = open(last_commit_hash_path, "r").read()
+
+        os.chdir(previous_folder)
+        return git_url, branch_name, last_commit_hash
+    else:
+        os.chdir(previous_folder)
+        kci_err("Not a GIT folder")
```

Comment:
We should also handle the case where the folder doesn't exist.

Reply:
already wrote a fix for this, is dealing also with different path for the repository
Trying to push the new commit but I have a problem with my nitrokey pro :cry:

Reply:
fixed by https://github.com/kernelci/kci-dev/pull/76/commits/3cfa957cf639c24e188e32a0abc3af56ff1ab4d8

Reply:
The behavior is following:
```
kci-dev results --git-folder arisut
Getting git repository informations for the current folder
git folder: arisut
Not a folder
Aborted!

```